### PR TITLE
[New Rule] Attempt to Mount an SMB Share via Command-line

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/913

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/lateral_movement_mounting_smb_share.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+[metadata]
+creation_date = "2021/01/25"
+maturity = "production"
+updated_date = "2021/01/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of macOS built-in commands to mount an SMB network share. Adversaries may use Valid Accounts to
+interact with a remote network share using Server Message Block (SMB).
```
Comment (bm11100) [inline rules/macos/lateral_movement_mounting_smb_share.toml:10]:
```suggestion
Identifies the execution of macOS built-in commands to mount a Server Message Block (SMB) network share. Adversaries may use valid accounts to
interact with a remote network share using SMB.
```

File: rules/macos/lateral_movement_mounting_smb_share.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+ (
+  process.name:"mount_smbfs" or
+  (process.name:"open" and process.args:"smb://*") or
+  (process.name:"mount" and process.args:"smbfs") or
+  (process.name:"osascript" and process.command_line:"osascript*mount volume*smb://*")
+ )
```
Comment (bm11100) [inline rules/macos/lateral_movement_mounting_smb_share.toml:31]:
```suggestion
process where event.type in ("start", "process_started") and
  (
    process.name : "mount_smbfs" or
    (process.name : "open" and process.args : "smb://*") or
    (process.name : "mount" and process.args : "smbfs") or
    (process.name : "osascript" and process.command_line : "osascript*mount volume*smb://*")
  )
```

Comment (bm11100) [reply rules/macos/lateral_movement_mounting_smb_share.toml:31]: just spacing for consistency, following jibs recommendations

File: rules/macos/lateral_movement_mounting_smb_share.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+author = ["Elastic"]
+description = """
+Identifies the execution of macOS built-in commands to mount an SMB network share. Adversaries may use Valid Accounts to
+interact with a remote network share using Server Message Block (SMB).
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Attempt to Mount an SMB Share via Command-line"
```
Comment (bm11100) [inline rules/macos/lateral_movement_mounting_smb_share.toml:16]:
```suggestion
name = "Attempt to Mount SMB Share via Command Line"
```

File: rules/macos/lateral_movement_mounting_smb_share.toml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2021/01/25"
+maturity = "production"
+updated_date = "2021/01/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of macOS built-in commands to mount a Server Message Block (SMB) network share. Adversaries may use valid accounts to
+interact with a remote network share using SMB.
```
Comment (bm11100) [inline rules/macos/lateral_movement_mounting_smb_share.toml:10]: this may need linting but otherwise LGTM!

**Review by bm11100 — APPROVED** [review]

**Review by threat-punter — APPROVED** [review]

**Pull request merged**